chapter_3/CMakeLists.txt

```diff
@@ -11,7 +11,9 @@ include_directories("${PROJECT_SOURCE_DIR}/libflandmark")
 FILE(COPY "haarcascade_frontalface_alt.xml" DESTINATION ${CMAKE_BINARY_DIR})
 FILE(COPY "flandmark_model.dat" DESTINATION ${CMAKE_BINARY_DIR})
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -stdlib=libc++")
+set(CMAKE_C_COMPILER "/usr/bin/clang")
+set(CMAKE_CXX_COMPILER "/usr/bin/clang++")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -stdlib=libc++ -stdlib=libstdc++")
```

Comment:
Then there should be a check on the compiler used, and depending on which compiler, the correct option should be given. Just changing it is rubbish, since CLANG is still not a default compiler on most systems but gcc and g++ are.